views/js/previewer/plugins/tools/scale/scale.js

```diff
@@ -0,0 +1,532 @@
+                sizeSettings = {
+                    width: val[1],
+                    height: val[0]
+                };
+            }
+            else {
+                sizeSettings = {
+                    width: val[0],
+                    height: val[1]
+                };
+            }
+
+            if (sizeSettings.width === container.width() && sizeSettings.height === container.height()) {
+                return false;
+            }
+
+            container.css(sizeSettings);
+            this.scaleFrame();
+        },
+
+        /**
+         * Configures selected device's orientation configuration state according to passed value from affected control
+         * and calls this.setOrientation(newOrientation) and this.scaleFrame() after all.
+         * @param {Event} event - event object related to affected control.
+         * @private
+         */
+        onOrientationChange: function onOrientationChange(event) {
+            var element = event.target;
+            var container = controls.$scaleWrapper.find('.' + selectedDevice + '-preview-container');
+            var newOrientation = element.value;
+
+            if (newOrientation === orientation) {
+                return false;
+            }
+            sizeSettings = {
+                height: container.width(),
+                width: container.height()
+            };
+
+            container.css(sizeSettings);
+
+            this.setOrientation(newOrientation);
+            this.scaleFrame();
+        },
+        
+        /**
+         * Configures selected device type on which depends all other parts of the plugin
+         * decides whether it needs to draw device frame around item content and scale it accordingly
+         * @param {Event} event - event object related to affected control.
+         * @private
+         */
+        onDeviceTypeChange: function onDeviceTypeChange(event) {
+            var element = event.target;
+
+            selectedDevice = element.value;
+
+            this.composeControlsByDeviceType(element.value);
+
+            if(element.value === DEFAULT_TYPE){
+                this.removeDeviceFrame();
+            }else{
+                this.changeDeviceFrame(element.value, true);
+                this.scaleFrame();
+                this.setupScreenSize();
+                if (element.value === 'mobile'){
+                    controls.$mobileDevices.children('.mobile-device-selector').get(0).dispatchEvent(new Event('change'));
+                } else if( element.value === 'desktop'){
+                    controls.$desktopDevices.get(0).dispatchEvent(new Event('change'));
+                }
+
+            }
+        }
+
+    };
+
+    return pluginFactory({
+
+        name: 'scale',
+
+        /**
+         * Initialize the plugin (called during runner's init)
+         */
+        init: function init() {
+            var self = this;
+            testRunner = this.getTestRunner();
+
+
+            controls = this.controls = {
```

Comment:
This should not happen. Module scope variables should not be used for instance purpose.